app/organisation/rest.py

```diff
@@ -262,6 +263,24 @@ def update_organisation_letter_branding_pool(organisation_id):
     return {}, 204
 
 
+@organisation_blueprint.route(
+    "/<uuid:organisation_id>/letter-branding-pool/<uuid:letter_branding_id>", methods=["DELETE"]
+)
+def remove_letter_branding_from_organisation_pool(organisation_id, letter_branding_id):
+    organisation = dao_get_organisation_by_id(organisation_id)
+    letter_branding_ids = {branding.id for branding in organisation.letter_branding_pool}
+
+    if letter_branding_id not in letter_branding_ids:
+        raise InvalidRequest(f"Letter branding {letter_branding_id} not in {organisation.name}'s pool", status_code=404)
+
+    if organisation.letter_branding_id == letter_branding_id:
+        raise InvalidRequest("You cannot remove an organisation's default letter branding", status_code=400)
+
+    dao_remove_letter_branding_from_organisation_pool(organisation_id, letter_branding_id)
```

Comment:
Some thoughts only, nothing to do here:

I think I need to understand better what these dao functions are for and how we want to be using them. We call `dao_get_organisation_by_id` at the start of this view, but then pass the organisation_id into this function which immediately calls `dao_get_organisation_by_id` again. And `dao_get_organisation_by_id` returns a SQLAlchemy model rather than plain data, so we're not hiding much away and still providing something that has direct access to the DB. Which means that on line 271 when we access `organisation.letter_branding_pool`, we are making more DB calls (perhaps unexpectedly).

Maybe one to discuss in the next tech chat ... as I think our abstraction layer here might not be hiding as much as we would hope.

Reply:
I agree, that sounds like a good topic to discuss at the tech catch up